module/Motion/CorrectingRotation.cpp

```diff
@@ -0,0 +1,121 @@
+/**
+ * @file   CorrectingRotation.h
+ * @brief  角度補正回頭
+ * @author aridome222
+ */
+
+#include "CorrectingRotation.h"
+
+using namespace std;
+
+CorrectingRotation::CorrectingRotation(int _targetAngle, double _targetSpeed)
+  : targetAngle(_targetAngle), targetSpeed(_targetSpeed){};
+
+void CorrectingRotation::run()
+{
+  // 事前条件を判定する
+  if(!isMetPrecondition()) {
+    return;
+  }
+
+  /*
+   * rear_camera_request.shより直線の角度を取得
+   * シミュレータ環境ではpopenが使えないため，
+   * ファイルに出力して読み込む
+   */
+  FILE* fp = nullptr;
+  char output[8];  // rear_camera_request.shの出力結果を保持する領域
+  char cmd[1024];
+  snprintf(cmd, 1024, "bash ./etrobocon2023/scripts/rear_camera_request.sh %d > response.txt",
+           ANGLE_SERVER_PORT);
+  system(cmd);
+  if((fp = fopen("response.txt", "r")) == NULL) {
+    // コマンドを実行できなかった場合Warningを出して終了する
+    logger.logWarning("Could not open \"./response.txt\"");
+    return;
+  }
+  // 実行結果をoutputにセット
+  fgets(output, sizeof(output), fp);
+  fclose(fp);
+
+  // outputから改行を削除
+  strcpy(output, StringOperator::removeEOL(output));
+
+  // 文字列がNoneの場合は黒線を認識できていないので，Warningを出して終了する
+  if(strcmp(output, "None") == 0) {
+    logger.logWarning("Could not recognize black line");
+    return;
+  }
+
+  double measuredAngle = atof(output);  // outputをdouble型に変換
+  int tAngle = targetAngle;
+
+  /*
+   * 目標角度0の時，縦線を取得できなかった場合は横線を採用する
+   * 計測角度(絶対値)が目標角度より45度より大きい場合，横線を採用したと判定する
```

Comment:
以下の変更で合ってるか分からないけど、コメントが変な気がします。
```suggestion
   * 計測角度と目標角度の差の絶対値が45度より大きい場合，横線を採用したと判定する
```

Reply:
ありがとうございます。日本語おかしいと思っていたのですが、修正するのを忘れてました。